### ISSUE TITLE
error: error executing az   message: ERROR: az: error: unrecognized arguments: --subscription

### ISSUE THREAD

**Issue by pranavtiwari09** [mspnp/template-building-blocks#414]
Hi,
I am working on AZ301 practice set: https://github.com/MicrosoftLearning/AZ-301-MicrosoftAzureArchitectDesign/blob/master/Instructions/AZ-301T04_Lab_Mod03_Deploying%20Network%20Infrastructure%20for%20use%20in%20Azure%20Solutions.md
There is no parameter in hub-vnet.json file. Also, the error keep happening no matter which .json file I use in this module.

I am on the latest Azure CLI and have installed azbb package using azure CLI. azbb Version is 2.2.1
Tried added subscription extension. Just doesn't work.
Please help - I am stuck.

I am running following command:
azbb -g AADesignLab08-hub-rg -s 3a25883a-xxxx-xxxx-xxxx-f2a11b530037 -l westus -p ~/hub-vnet.json --deploy

Keep getting error: error: error executing az
  message: ERROR: az: error: unrecognized arguments: --subscription

** Even if I remove everything and just keep this "azbb -p ~/hub-vnet.json --deploy" still get the same error. Looks like something is wrong with schema file. But I am not that advanced user - so don't know for sure.

**Comment by atoakley** [comment]
@pranavtiwari09 The error you received is coming from the AZ CLI. Please run "az --version" and make sure you are running a current version of the AZ CLI. You can also run "az group create --help", and see if there is a global argument named "subscription".

I was able to successfully run your command and deploy the hub virtual network.

**Comment by thy143** [comment]
I am also experancing the same issue.  I have form a fresh install of Windows and also from a fresh install of Linux, getting a similar error when trying to run azbb.



**Error:**

```
error: error executing az
  message: ERROR: az: error: unrecognized arguments: --subscription <sub-id>
usage: az [-h] [--verbose] [--debug]
          [--output {json,jsonc,table,tsv,yaml,none}] [--query JMESPATH]
          {account} ...
  status: 2
  arguments: account list-locations --subscription <sub-id> --query "[].name | sort(@)"
 
 ```

The error occures when azbb calls `az account list-locations --subscription`. When performing az account list-locations --help` i see that --subscription is not a valid parameter.


**azbb version**

```

>azbb -V
2.2.1
 ```
**az version:**

```
>az -v
azure-cli                         2.0.71
 
command-modules-nspkg               2.0.3
core                              2.0.71
nspkg                              3.0.4
telemetry                          1.0.3
 
Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\<username>\.azure\cliextensions'
 
Python (Windows) 3.6.6 (v3.6.6:4cf1f54eb7, Jun 27 2018, 02:47:15) [MSC v.1900 32 bit (Intel)]
 
Legal docs and information: aka.ms/AzureCliLegal
 
 
Your CLI is up-to-date.

**Comment by atoakley** [comment]
@pranavtiwari09 @thy143 I have merged PR #415 that should resolve this. I have published it to npm, so please upgrade and let me know if it works for you.
